File: app/src/main/java/com/musicplayer/SocyMusic/ui/sleeptime/SleepTimeActivity.java
package com.musicplayer.SocyMusic.ui.sleeptime;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.musicplayer.musicplayer.R;

public class SleepTimeActivity extends AppCompatActivity {
    private SleepTimeFragment sleepTimeFragment;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sleeptime);
        setTitle("Select Sleeptime");

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentById(R.id.layout_settings_container);
        if (fragment == null) {
            sleepTimeFragment = new SleepTimeFragment();
            fragmentManager.beginTransaction()
                    .add(R.id.layout_sleeptime_container, sleepTimeFragment)
                    .commit();
        } else {
            sleepTimeFragment = (SleepTimeFragment) fragment;
        }
    }

    @Override
    public void onBackPressed() {
        sleepTimeFragment.onBackPressed();
    }
}

File: app/src/main/kotlin/com/musicplayer/openmusic/ui/sleeptime/SleepTimeActivity.kt
package com.musicplayer.openmusic.ui.sleeptime

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.musicplayer.openmusic.R

class SleepTimeActivity : AppCompatActivity() {
    private var sleepTimeFragment: SleepTimeFragment? = null
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sleeptime)
        title = "Select Sleeptime"
        val fragmentManager = supportFragmentManager
        val fragment = fragmentManager.findFragmentById(R.id.layout_settings_container)
        if (fragment == null) {
            sleepTimeFragment = SleepTimeFragment()
            fragmentManager.beginTransaction()
                .add(R.id.layout_sleeptime_container, sleepTimeFragment!!)
                .commit()
        } else {
            sleepTimeFragment = fragment as SleepTimeFragment
        }
    }

    override fun onBackPressed() {
        sleepTimeFragment!!.onBackPressed()
    }
}